src/concerttickets/models/tickets/Ticket.java

```diff
@@ -0,0 +1,220 @@
+package concerttickets.models.tickets;
+
+import concerttickets.models.contactservices.Email;
+import concerttickets.models.contactservices.Phone;
+import concerttickets.models.Template;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+//import java.util.Random;
+
+public class Ticket extends Template {
+    private String hall;
+    private int code;
+    private long time;
+    private boolean isPromo;
+    private Sector sector;
+    private BigDecimal maxWeight;
+    private BigDecimal price;
+
+    private LocalDateTime creationDate;
+    static int lastId = 1000;
+
+
+    public Ticket(String hall, int code, LocalDateTime time, boolean isPromo, Sector sector,
+                  double maxWeight, double price) {
+        this.createFullTicket(hall, code, time, isPromo, sector, maxWeight, price);
+
+    }
+
+    public Ticket(String hall, int code, LocalDateTime time) {
+        this.createLimitedTicket(hall, code, time);
+    }
+
+    public Ticket() {
+
+    }
+
+    /*
+    private int generateId() {
+        StringBuilder newId = new StringBuilder();
+        Random random = new Random();
+        String characters = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+        for (int i = 0; i < 4; i++) {
+            newId.append(characters.charAt(random.nextInt(characters.length())));
+        }
+        return newId.toString();
+    }
+    */
+    private void createLimitedTicket(String hall, int code, LocalDateTime time) {
+        this.setId(Ticket.lastId++);
```

Comment:
why does the id start from 1000?

Reply:
I probably misunderstood one of the previous home tasks with four digits or letters ID